R/dummy_multi_choice.R

```diff
@@ -157,16 +157,19 @@ prep.step_dummy_multi_choice <- function(x, training, info = NULL, ...) {
   )
 }
 
-multi_dummy_check_type <- function(dat) {
+multi_dummy_check_type <- function(dat, call = rlang::caller_env()) {
   is_good <- function(x) {
     is.factor(x) | is.character(x) | all(is.na(x))
   }
 
   all_good <- vapply(dat, is_good, logical(1))
   if (!all(all_good)) {
-    rlang::abort(
-      "All columns selected for the step should be factor, character, or NA"
-    )
+    offenders <- names(dat)[!all_good]
+    cli::cli_abort(c(
+      "x" = "All columns selected for the step should be \\
+            factor, character, or NA. The following were not:",
+      "*" = "{.and {.var {offenders}}}."
```

Comment:
```suggestion
      "*" = "{.var {offenders}}."
```

`.and` markup is redundant.